---
title: "PRS veronique"
author: "Véronique Boumtje. Some modifcations and plotting adjustemnts by Sébastien Renaut"
date: "`r Sys.Date()`"
output:
  html_document:
    number_sections: F
params:
  datapath: 'C:/Users/renseb01/Documents/PRS'
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = params$datapath)
knitr::opts_chunk$set(echo = F)
library(dplyr)
library(tidyr)
library(ggplot2)
library(patchwork)
library(caret)
library(rms)
library(psychometric)
library(pROC)
library(RColorBrewer)
library(ggsignif)
library(gridExtra)
```


```{r data upload, include=T,eval=T}
PRS_litt <- read.csv(file.path(params$datapath,'data/Table_ClinData_PRS_GRS_20231214.csv'))
###SNP as number of alleles
PRS_litt$rs1051730_alleles = 0
PRS_litt$rs1051730_alleles[PRS_litt$rs1051730=='AA']  = 2
PRS_litt$rs1051730_alleles[PRS_litt$rs1051730=='GA' ]  =  1
PRS_litt$rs1051730_alleles[PRS_litt$rs1051730=='GG'] =  0
PRS_litt$rs1051730_alleles = as.numeric(PRS_litt$rs1051730_alleles)
```


```{r regression models litterature, include=FALSE}
glm_models = NULL

values = c("PRS_LORD_CaG","GRS_Bossé_44_unweighted","GRS_Bossé_44_weighted","GRS_Byun_38_unweighted",
"GRS_Byun_38_weighted","GRS_Bossé_9_unweighted","GRS_Bossé_9_weighted",
"GRS_Byun_13_unweighted","GRS_Byun_13_weighted","GRS_Dai_19_unweighted","GRS_Dai_19_weighted","GRS_Shi_6_unweighted",
"GRS_Shi_6_weighted","GRS_Graff_102_unweighted","GRS_Graff_102_weighted","GRS_Jia_19_unweighted","GRS_Jia_19_weighted",
"GRS_Fritsche_14_unweighted","GRS_Fritsche_14_weighted","GRS_Fritsche_19_unweighted","GRS_Fritsche_19_weighted","GRS_Hung_35_unweighted",
"GRS_Hung_35_weighted","GRS_Hung_125_unweighted","GRS_Hung_125_weighted","GRS_Zhang_32_unweighted","GRS_Zhang_32_weighted",
"rs1051730_alleles")

categories =  c("Genome-wide",rep("GWAS-SNP",8),rep("Literature",18),"sentinel SNP")

#calculate GLMs
for(i in seq_along(values))
  {
  #data
  data = PRS_litt[,colnames(PRS_litt) %in% c('Lung_cancer',values[i])]
  
  #Z score normalisation
  if(length(grep('weight',colnames(data)[2]))==1) data[,2] <- (data[,2]-mean(data[,2]))/sd(data[,2])
  if(length(grep('rs1051730_alleles',colnames(data)[2]))==1) data[,2] <- (data[,2]-mean(data[,2]))/sd(data[,2])
  
  #regression models
  regression_model <- glm(Lung_cancer ~ . , data = data, family = binomial)
  
  #OR
  model_table <- broom::tidy(regression_model,conf.int = T)
  model_table = model_table[model_table$term!="(Intercept)",]
  model_table <- model_table %>% mutate(odds_ratio = exp(estimate))
  
  #LRM 
  lrm_model <- rms::lrm(Lung_cancer~., data=data)
  
  model_table$r2 = lrm_model$stats[c("R2")]
  model_table$r2_LCL = CI.Rsq(model_table$r2,n=24012,k=1)$LCL
  model_table$r2_UCL = CI.Rsq(model_table$r2,n=24012,k=1)$UCL

  model_table$AUC =  lrm_model$stats[c("C")]
  model_table$PRS = gsub('_weighted','',model_table$term)
  model_table$PRS = gsub('_unweighted','',model_table$PRS)
  model_table$PRS = ifelse(model_table$PRS=='rs1051730_alleles','rs1051730',model_table$PRS)
  model_table$PRS = ifelse(model_table$PRS=='PRS_LORD_CaG','PRS_Genome-wide',model_table$PRS)
  
  model_table$PRS = gsub('^.RS_','',model_table$PRS)
  model_table$Type = ifelse(length(grep('_weighted',model_table$term)),'weighted','unknown')
  model_table$Type = ifelse(length(grep('_unweighted',model_table$term)),'unweighted',model_table$Type)
  model_table$Type[model_table$PRS== 'Genome-wide'] = 'weighted'
  model_table$Type[model_table$PRS== 'rs1051730'] = 'unweighted'
  
  model_table$categories = categories[i]
  
  #95% CI for OR
  model_table$ymin_OR = exp(model_table$estimate - (1.96 * model_table$std.error))
  model_table$ymax_OR = exp(model_table$estimate + (1.96 * model_table$std.error))
  
  #final cbind...
  glm_models = rbind(glm_models,model_table)
  
  print(paste0('Done ',i,': ', colnames(data)[2]))
}

write.csv(glm_models,file.path(params$datapath,'results/glm_models.csv'),row.names = F)
```


```{r regression models per quintile, include=FALSE}
####calcul du Odd Ratio par quintile
#PRS_litt$PRS_quintile <- with(PRS_litt, cut(PRS_LORD_CaG, breaks = quantile(PRS_LORD_CaG,probs =seq(0,1,by=0.2),na.rm = TRUE), labels= c("1","2","3","4","5"), include.lowest= TRUE))
PRS_litt$PRS_quintile <- with(PRS_litt, cut(PRS_LORD_CaG, breaks = quantile(PRS_LORD_CaG,probs = c(0,0.8,0.9,0.95,0.99),na.rm = TRUE), labels= c("1","2","3","4"), include.lowest= TRUE))

PRS_litt$PRS_quintile <- relevel(PRS_litt$PRS_quintile, "1")

model.PRS <- glm(Lung_cancer ~ PRS_quintile, data = PRS_litt, family = binomial, na.action = na.omit)

model.PRS_table <- broom::tidy(model.PRS)
model.PRS_table$estimate <- ifelse(model.PRS_table$term == "(Intercept)",0, model.PRS_table$estimate)
  model.PRS_table <- model.PRS_table %>%
    mutate(odds_ratio = exp(estimate))
  
#model.PRS_table$std.error <- ifelse(model.PRS_table$term == "(Intercept)", NA, model.PRS_table$std.error)
model.PRS_table$term  <- ifelse(model.PRS_table$term == "(Intercept)",1, model.PRS_table$term)

#95% CI for OR
model.PRS_table$ymin_OR = exp(model.PRS_table$estimate - (1.96 * model.PRS_table$std.error))
model.PRS_table$ymax_OR = exp(model.PRS_table$estimate + (1.96 * model.PRS_table$std.error))

#save data
write.csv(model.PRS_table,file.path(params$datapath,'results/model.PRS_table.csv'),row.names = F)
```



```{r regression models per smoking, genetic risk and combined}
####calcul du Odd Ratio par bottom 20, 20-80, top 20 ET par status tabagique ET combiné) 
#Data_PRS_GRS_rmNAs = Data_PRS_GRS[!is.na(Data_PRS_GRS$tabagism),]
PRS_litt_df = as.data.frame(PRS_litt)
quantile = quantile(PRS_litt_df$PRS_LORD_CaG,c(0,0.2,0.8,1))

#Smoking
PRS_litt_df$tabagism_recoded = NA
PRS_litt_df$tabagism_recoded[PRS_litt_df$tabagism == "Current_smoker"] = 'Ever'
PRS_litt_df$tabagism_recoded[PRS_litt_df$tabagism == "Former_smoker"] = 'Ever'
PRS_litt_df$tabagism_recoded[PRS_litt_df$tabagism == "Never_smoker"] = 'Never'
PRS_litt_df$tabagism_recoded = factor(PRS_litt_df$tabagism_recoded,levels= c('Never','Ever'))

#Genetic Risk
PRS_litt_df$genetic_risk_factor = 'Low'
PRS_litt_df$genetic_risk_factor[PRS_litt_df$PRS_LORD_CaG > quantile[2] & PRS_litt_df$PRS_LORD_CaG <= quantile[3]] = 'Intermediate'
PRS_litt_df$genetic_risk_factor[PRS_litt_df$PRS_LORD_CaG > quantile[3]] = 'High'
PRS_litt_df$genetic_risk_factor = factor(PRS_litt_df$genetic_risk_factor, levels = c('Low','Intermediate','High'))

#Combined effect                      
PRS_litt_df$genetic_risk_factor_tabagism = paste0(PRS_litt_df$genetic_risk_factor,' ',PRS_litt_df$tabagism_recoded)       
PRS_litt_df$genetic_risk_factor_tabagism = factor(PRS_litt_df$genetic_risk_factor_tabagism, levels = c("Low Never","Low Ever","Intermediate Never","Intermediate Ever","High Never","High Ever"))

#Perform GLMs below
models_ALL =list()
variables = c('genetic_risk_factor_tabagism','genetic_risk_factor','tabagism_recoded')
smoking = c('Never','Ever')
genetic_risk_factor = c('Low','Intermediate','High')

for(c in 1:3){
    models = NULL
  
  #GLM Low, Intermediate, High
  if(c == 1) {
    for(i in 1:2) {
     #subset data
     PRS_litt_df_subset = PRS_litt_df[PRS_litt_df$tabagism_recoded == smoking[i],]
     model <- glm(Lung_cancer ~ genetic_risk_factor, data = PRS_litt_df_subset, family = binomial, na.action = na.omit)
     model <- broom::tidy(model,conf.int =T)
   
     #clean up 
     model$estimate <- ifelse(model$term == "(Intercept)",0, model$estimate)
    # model$std.error <- ifelse(model$term == "(Intercept)", NA, model$std.error)
     model$p.value <- ifelse(model$term == "(Intercept)",0, model$p.value)
     model$odds_ratio = exp(model$estimate)
     model$Type = c('Low','Intermediate','High')
     model$Smoking = smoking[i]
     model$N = c(table(PRS_litt_df_subset$genetic_risk_factor))
     
     
     #95% CI for OR
     model$ymin_OR = exp(model$estimate - (1.96 * model$std.error))
     model$ymax_OR = exp(model$estimate + (1.96 * model$std.error))
     
     #save
     models =rbind(models,model)
     if(i==2) {
       models$Type = factor(models$Type,level = c('Low','Intermediate','High'))
       models$Smoking = factor(models$Smoking,level = c('Never','Ever'))
       models_ALL[[c]] = models
       } 
    }
  }  
  
  #GLMs Never/Ever 
  if(c == 2) {
    for(i in 1:3) {
     #subset data
     PRS_litt_df_subset = PRS_litt_df[PRS_litt_df$genetic_risk_factor == genetic_risk_factor[i],]
     model <- glm(Lung_cancer ~ tabagism_recoded, data = PRS_litt_df_subset, family = binomial, na.action = na.omit)
     model <- broom::tidy(model,conf.int =T)
   
     #clean up
     model$estimate <- ifelse(model$term == "(Intercept)",0, model$estimate)
    # model$std.error <- ifelse(model$term == "(Intercept)", NA, model$std.error)
     model$p.value <- ifelse(model$term == "(Intercept)",0, model$p.value)
     model$odds_ratio = exp(model$estimate)
     model$Type = c('Low','Intermediate','High')[i]
     model$Smoking = c('Never','Ever')
     model$N = c(table(PRS_litt_df_subset$tabagism_recoded))
     

     #95% CI for OR
     model$ymin_OR = exp(model$estimate - (1.96 * model$std.error))
     model$ymax_OR = exp(model$estimate + (1.96 * model$std.error))
     
     #save
     models =rbind(models,model)
     if(i==3) {
       models$Type = factor(models$Type,level = c('Low','Intermediate','High'))
       models$Smoking = factor(models$Smoking,level = c('Never','Ever'))
       models_ALL[[c]] = models
       }
    }
  }
  
  #GLM combined
  if(c == 3) {
    model <- glm(Lung_cancer ~ genetic_risk_factor_tabagism, data = PRS_litt_df, family = binomial, na.action = na.omit)
    model <- broom::tidy(model,conf.int =T)

    #clean up 
    model$estimate <- ifelse(model$term == "(Intercept)",0, model$estimate)
   # model$std.error <- ifelse(model$term == "(Intercept)", NA, model$std.error)
    model$p.value <- ifelse(model$term == "(Intercept)",0, model$p.value)
    model$odds_ratio = exp(model$estimate)
    model$N = c(table(PRS_litt_df$genetic_risk_factor_tabagism))
    model$Type = factor(c('Low','Low','Intermediate','Intermediate','High','High'),level = c('Low','Intermediate','High'))
    model$Smoking = factor(rep(c('Never','Ever'),3),level = c('Never','Ever'))
    model$term = factor(c("Low ~ Never","Low ~ Ever","Intermediate ~ Never","Intermediate ~ Ever","High ~ Never","High ~ Ever"),level = c("Low ~ Never","Low ~ Ever","Intermediate ~ Never","Intermediate ~ Ever","High ~ Never","High ~ Ever"))

    #95% CI for OR
    model$ymin_OR = exp(model$estimate - (1.96 * model$std.error))
    model$ymax_OR = exp(model$estimate + (1.96 * model$std.error))
    
    #save
    models_ALL[[c]] = model
    }
}

#save
saveRDS(models_ALL,file.path(params$datapath,'results/models_ALL.RDS'))
```


```{r Figure 2 improved}
glm_models = read.csv(file.path(params$datapath,'results/glm_models.csv'))

#remove bosse_9 and buyn_13
glm_models =  glm_models[(glm_models$term %in% c('GRS_Bossé_9_weighted','GRS_Bossé_9_unweighted','GRS_Byun_13_unweighted','GRS_Byun_13_weighted')) == F,]

glm_models$categories[1]  = 'Genome-\nwide' 
glm_models$categories[24] = 'Sentinel\nSNP'
glm_models$categories = factor(glm_models$categories,levels= c('GWAS-SNP','Literature','Sentinel\nSNP','Genome-\nwide'))
glm_models$term = factor(glm_models$term, levels = c("PRS_Genome-wide","GRS_Bossé_44_unweighted","GRS_Bossé_44_weighted","GRS_Byun_38_unweighted",
"GRS_Byun_38_weighted","GRS_Dai_19_unweighted","GRS_Dai_19_weighted","GRS_Shi_6_unweighted",
"GRS_Shi_6_weighted","GRS_Graff_102_unweighted","GRS_Graff_102_weighted","GRS_Jia_19_unweighted","GRS_Jia_19_weighted",
"GRS_Fritsche_14_unweighted","GRS_Fritsche_14_weighted","GRS_Fritsche_19_unweighted","GRS_Fritsche_19_weighted","GRS_Hung_35_unweighted",
"GRS_Hung_35_weighted","GRS_Hung_125_unweighted","GRS_Hung_125_weighted","GRS_Zhang_32_unweighted","GRS_Zhang_32_weighted",
"rs1051730"))
glm_models$Type = factor(glm_models$Type,levels = c('unweighted','weighted'))
glm_models$PRS = factor(glm_models$PRS, levels = c("Genome-wide","Bossé_44","Byun_38","Dai_19","Shi_6", "Graff_102",  "Jia_19","Fritsche_14","Fritsche_19","Hung_35","Hung_125","Zhang_32","rs1051730"))


# OR
plots_or <- ggplot(glm_models, aes(x = PRS, y = odds_ratio,group = term,fill = Type)) +
  geom_bar(stat = "identity" , position='dodge', width = c(0.45,rep(0.9,22),0.45)) + 
  geom_errorbar(aes(ymin= ymin_OR, ymax= ymax_OR), width = .2, position = position_dodge(.9)) +
  geom_hline(yintercept=1,linetype="dashed") +
  scale_fill_brewer(palette="Paired") + 
  facet_grid(.~categories, scales = "free", space = "free") +
  coord_cartesian(ylim=c(1,1.4)) +
  labs(title = "", x= "PRS", y = "Odds ratio") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"),
        legend.position='none',
        plot.title = element_text(size=12, hjust = 0.5),
        axis.text.x = element_text(size = 8,angle = 45, hjust = 1),
        axis.title.y = element_text(size = 10),
        strip.text.x = element_text(size =9))

#PVE 
plots_pve <- ggplot(glm_models, aes(x = PRS, y = r2,group = term,fill = Type)) +
  geom_bar(stat = "identity" , position='dodge', width = c(0.45,rep(0.9,22),0.45)) + 
  geom_errorbar(aes(ymin= r2_LCL, ymax= r2_UCL), width = .2, position = position_dodge(.9)) +
  geom_hline(yintercept=1,linetype="dashed") +
  scale_fill_brewer(palette="Paired") + 
  facet_grid(.~categories, scales = "free", space = "free") +
  coord_cartesian(ylim=c(0.001,0.022)) +
  labs(title = "", x= "PRS", y = bquote('Variance Explained '(R^2))) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"),
        legend.position='right',
        plot.title = element_text(size=12, hjust = 0.5),
        axis.text.x = element_text(size = 8,angle = 45, hjust = 1),
        axis.title.y = element_text(size = 10),
        strip.text.x = element_text(size =9))


pdf(file.path(params$datapath,'results/Figure2.pdf'),width = 10,height = 10)
( plots_pve /plots_or ) + plot_annotation(tag_levels = 'A') +  plot_layout(guides = 'collect')
dev.off()
```


```{r FIGURE 3}
or_perquintile = read.csv(file.path(params$datapath,'results/model.PRS_table.csv'))
or_perquintile$labels = c('1','2','3','4','5')

#density plot
PRS_litt_df$Cohort <- factor(PRS_litt_df$Lung_cancer, levels = c(1,0), labels = c("LORD", "CARTaGENE"))
PRS_means = PRS_litt_df %>% group_by(Cohort) %>% summarise(grp.means = mean(PRS_LORD_CaG) )

density_plots = ggplot(PRS_litt_df, aes(x=PRS_LORD_CaG,fill = Cohort)) + 
  geom_density(alpha = 0.4) +
  geom_vline(data=PRS_means , aes(xintercept=grp.means, color=Cohort),
             linetype="dashed", show.legend = F) + 
  xlab("PRS") +
  ylab("Density") +
  theme(legend.position = c(0.75, 0.2),
        legend.key.size = unit(0.1, 'cm'), #change legend key size
        legend.key.height = unit(0.2, 'cm'), #change legend key height
        legend.key.width = unit(0.2, 'cm'), #change legend key width
        legend.title = element_text(size=8), #change legend title font size
        legend.text = element_text(size=6),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"),
        legend.background = element_rect(fill = "gray75"),
        legend.key = element_rect(fill = "gray75", color = NA))


quintile_plot <- ggplot(or_perquintile, aes(x = labels, y = odds_ratio,group=1)) + 
  geom_errorbar(aes(ymin= ymin_OR, ymax= ymax_OR), width = .2, position = position_dodge(.9)) +
  geom_point(stat = "identity",color='black',size = 3) +
  geom_hline(yintercept=1,linetype="dashed") + 
  xlab("Quintiles of PRS") +
  ylab("Odds ratio") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"))


# Regression model
model.Gw_PRS <- glm(Lung_cancer ~ PRS_LORD_CaG , data = PRS_litt_df, family = binomial,  na.action = na.omit)
# Get predicted probabilities for test data
PRS_litt_df$predict  <-  predict(model.Gw_PRS, PRS_litt_df, type="response")
# Plot ROC curve
P4 <- roc(PRS_litt_df$Lung_cancer, PRS_litt_df$predict, ci=TRUE, boot.n=1000, ci.alpha=0.9, stratified=FALSE,plot=FALSE, auc.polygon=TRUE, max.auc.polygon=TRUE, grid=FALSE,print.auc=TRUE, show.thres=TRUE,quiet= TRUE)
text_annotation = paste0('AUC: ',round(P4$ci[2],3),'0 (',round(P4$ci[1],3),'0-',round(P4$ci[3],3),')')

roc_curve = ggroc(P4) +
  geom_segment(aes(x = 1, y = 0, xend = 0, yend = 1),show.legend = F, linetype = 'dashed') +   ylab('Sensitivity') + xlab('Specificity') +
  annotate(geom="text", x=0.35, y=0.2, label=text_annotation) +
  theme(legend.position = c(0.8, 0.8),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"))

#table
table = read.delim(file.path(params$datapath,'data/Number_PRS_quantile1.txt'), check.names = F,row.names = 1)

pdf(file.path(params$datapath,'results/Figure3.pdf'),width = 10,height = 6)
((density_plots | roc_curve | quintile_plot) / tableGrob(table)) + plot_layout(height = c(4,2)) + plot_annotation(tag_levels = 'A')
dev.off()

###sup figure for interaction
pdf(file.path(params$datapath,'results/FigureSup_interaction.pdf'),width = 7.5,height = 7)
interaction.plot(x.factor = PRS_litt_df$genetic_risk_factor, #x-axis variable
                 trace.factor = PRS_litt_df$tabagism_recoded, #variable for lines
                 response = PRS_litt_df$Lung_cancer, #y-axis variable
                 fun = mean, #metric to plot
                 ylab = "Percentage of patients with lung cancer",
                 xlab = "genetic risk factor",
                 col = c("red", "blue"),
                 lty = 1, #line type
                 lwd = 3, #line width
                 trace.label = "Smoking status",pch = 15,type='b')
dev.off()
```


```{r FIGURE 4}
models_ALL = readRDS(file.path(params$datapath,'results/models_ALL.RDS'))

#plots
plots_all <- ggplot(models_ALL[[3]],aes(x = Type, y= odds_ratio,fill = term)) + 
  geom_bar(stat = "identity",position = position_dodge(),width = 0.6) + 
  geom_errorbar(aes(ymin= ymin_OR, ymax= ymax_OR), width = .2, position = position_dodge(.6)) +
  geom_hline(yintercept=1,linetype="dashed") +
  scale_fill_manual('Combined effect',values=brewer.pal(8,'Paired')[c(1:4,7:8,6)],
                    labels = c("Low ~ Never","Low ~ Ever","Interm. ~ Never","Interm. ~ Ever","High ~ Never","High ~ Ever")) +
  geom_label(aes(label=N,y=-2), vjust=0.4, color="gray10",label.size=0,
               position = position_dodge(0.6),show.legend = F) +
   geom_label(aes(x = c(0.85,1.15,1.85,2.15,2.85,3.15), y = odds_ratio, label = round(odds_ratio,2)),label.size = 0,show.legend = F,vjust = -0.5, inherit.aes = F,label.padding =unit(0.1, "lines")) +
  ylim(c(-2,25)) +
  geom_signif(y_position =  max(models_ALL[[3]]$ymax_OR) + 1, xmin = 0.85, xmax = 3.15,
    annotation = c("OR = 17.0X***"), tip_length = 0.01) +
  labs(title = "", x= "Combination of genetic risk and smoking subgroups", y = "Odds ratio") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"),
        legend.position='right') #,


#smoking effect

plots_smoking <- ggplot(models_ALL[[2]],aes(x = Type, y= odds_ratio,fill = Smoking)) + 
  geom_bar(stat = "identity" , position = position_dodge(),width = 0.6) + 
  geom_errorbar(aes(ymin= ymin_OR, ymax= ymax_OR), width = .2, position = position_dodge(0.6)) +
  geom_hline(yintercept=1,linetype="dashed") +
  scale_fill_manual('Smoking',values=brewer.pal(11,'BrBG')[c(5,1)],
                    labels = c("Never","Ever")) +
  geom_label(aes(label=N,y=-1), vjust=0.4, color=rep(c("gray10","gray90"),3),label.size=0,
               position = position_dodge(0.6),show.legend = F) +
  geom_signif(y_position = rep(max(models_ALL[[2]]$ymax_OR) + 0.5,3), xmin = c(0.85,1.85,2.85), xmax = c(1.15,2.15,3.15),
    annotation = c("OR = 7.7X***","OR = 9.8X***","OR = 11.3X***"), tip_length = 0.01) +
  ylim(c(-1,16)) +
  labs(title = "", x= "Genetic risk groups", y = "Odds ratio") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"),
        legend.position='right') #,

#quantile effect

plots_quintile <- ggplot(models_ALL[[1]],aes(x = Smoking, y= odds_ratio,fill = Type)) + 
  geom_bar(stat = "identity" , position = position_dodge(),width = 0.6) + 
  geom_errorbar(aes(ymin= ymin_OR, ymax= ymax_OR), width = .2, position = position_dodge(0.6)) +
  geom_hline(yintercept=1,linetype="dashed") +
  scale_fill_manual('Genetic Risk',values=brewer.pal(11,'PRGn')[c(5,3,1)],
                    labels =  c("Low","Intermediate","High")) +
  geom_label(aes(label=N,y=-0.2), vjust=0.4,color=rep(c("gray10","gray10","gray90"),2),label.size=0,
               position = position_dodge(0.6),show.legend = F) +
  geom_signif(y_position = rep(max(models_ALL[[1]]$ymax_OR) + 0.1,2), xmin = c(0.8,1.8), xmax = c(1.2,2.2),
    annotation = c("OR = 1.5X*","OR = 2.2X***"), tip_length = 0.01) +
  ylim(c(-0.2,2.6)) +
  labs(title = "", x= "Smoking groups", y = "Odds ratio") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black"),
        legend.position='right') #,


pdf(file.path(params$datapath,'results/Figure4.pdf'),width = 8,height = 12)
(plots_quintile / plots_smoking / plots_all + plot_annotation(tag_levels = 'A'))  & theme(legend.justification = "left")
dev.off()
```



# session info  
```{r session, message= T}
###session
sessionInfo()


PRS_litt_df = as.data.frame(PRS_litt)
quantile = quantile(PRS_litt_df$PRS_LORD_CaG,c(0,0.2,0.8,1))

#Smoking
PRS_litt_df$tabagism_recoded = NA
PRS_litt_df$tabagism_recoded[PRS_litt_df$tabagism == "Current_smoker"] = 'Ever'
PRS_litt_df$tabagism_recoded[PRS_litt_df$tabagism == "Former_smoker"] = 'Ever'
PRS_litt_df$tabagism_recoded[PRS_litt_df$tabagism == "Never_smoker"] = 'Never'
PRS_litt_df$tabagism_recoded = factor(PRS_litt_df$tabagism_recoded,levels= c('Never','Ever'))

#Genetic Risk
PRS_litt_df$genetic_risk_factor = 'Low'
PRS_litt_df$genetic_risk_factor[PRS_litt_df$PRS_LORD_CaG > quantile[2] & PRS_litt_df$PRS_LORD_CaG <= quantile[3]] = 'Intermediate'
PRS_litt_df$genetic_risk_factor[PRS_litt_df$PRS_LORD_CaG > quantile[3]] = 'High'
PRS_litt_df$genetic_risk_factor = factor(PRS_litt_df$genetic_risk_factor, levels = c('Low','Intermediate','High'))

#Combined effect                      
PRS_litt_df$genetic_risk_factor_tabagism = paste0(PRS_litt_df$genetic_risk_factor,' ',PRS_litt_df$tabagism_recoded)       
PRS_litt_df$genetic_risk_factor_tabagism = factor(PRS_litt_df$genetic_risk_factor_tabagism, levels = c("Low Never","Low Ever","Intermediate Never","Intermediate Ever","High Never","High Ever"))

#Perform GLMs below
models_ALL =list()
variables = c('genetic_risk_factor_tabagism','genetic_risk_factor','tabagism_recoded')
smoking = c('Never','Ever')
genetic_risk_factor = c('Low','Intermediate','High')

#recode variables as numeric....
PRS_litt_df$genetic_risk_factor <-   car::recode(PRS_litt_df$genetic_risk_factor, "'Low'= 0; 'Intermediate'=1; 'High'=2",as.factor=F)
PRS_litt_df$Smoking_status <- car::recode(PRS_litt_df$tabagism_recoded, "'Never'=0; 'Ever'=1",as.factor=F)
model.interaction_group1<- glm(Lung_cancer ~ Smoking_status*genetic_risk_factor, data = PRS_litt_df, family = binomial, na.action = na.omit)

#LRT



model.tab.clin <- glm(Lung_cancer ~ Age + Sex + BMI + tabagism + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10, data = PRS_litt_df, family = binomial,  na.action = na.omit)

model.Gw_PRS.tab.clin <- glm(Lung_cancer ~ PRS_LORD_CaG + Age + Sex + BMI + tabagism + PC1 + PC2 + PC3 + PC4 + PC5 + PC6 + PC7 + PC8 + PC9 + PC10, data = PRS_litt_df, family = binomial,  na.action = na.omit)

m1 <- glm(Lung_cancer ~  tabagism, data = PRS_litt_df, family = binomial,  na.action = na.omit)
m2 <- glm(Lung_cancer ~ Sex  + tabagism, data = PRS_litt_df, family = binomial,  na.action = na.omit)
m3 <- glm(Lung_cancer ~ Sex , data = PRS_litt_df, family = binomial,  na.action = na.omit)


lmtest::lrtest(model.tab.clin, model.Gw_PRS.tab.clin)
#Voici le code de Yohan(je trouve exactement le même résultat que le précédent)
anova(model.tab.clin, model.Gw_PRS.tab.clin, test="LRT")
 



```


